content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md

```diff
@@ -0,0 +1,382 @@
+---
+title: "Handling application performance monitoring on HPE GreenLake for Private
+  Cloud Enterprise – Part 2: App monitoring using Apache SkyWalking"
+date: 2023-01-04T07:31:55.711Z
+author: Guoping Jia
+authorimage: /img/guoping.png
+tags:
+  - hpe-greenlake, hpe-greenlake-for-private-cloud-enterprise, Kubernetes,
+    application performance monitoring, Apache SkyWalking
+---
+## Introduction
+
+[HPE GreenLake for Private Cloud Enterprise](https://www.hpe.com/us/en/greenlake/private-cloud-enterprise.html) delivers a modern private cloud to support your app workloads with bare metal, containers, and virtual machines (VMs) running in any combination across your edges, colocations, and data centers. It combines self-service resource access for developers with consumption and performance transparency for IT. Within this modern application environment, having a robust application performance monitoring (APM) tool is becoming essential. It can help IT professionals to ensure that deployed applications meet the performance, reliability and valuable user experience required by developers, partners and customers.
+
+In [the first blog post](https://developer.hpe.com/blog/get-started-with-application-performance-monitoring-tools-overview/), I walked through some of the best APM tools, and I described their key features and discussed in details their strengths and weaknesses.
+
+In this blog post, I will start choosing one APM tool,  *Apache SkyWalking*, and describe the detailed process how to set it up in HPE GreenLake for Private Cloud Enterprise for monitoring and alerting customer applications. 
+
+## Apache SkyWalking
+
+[Apache SkyWalking](https://skywalking.apache.org/) is an open source APM tool with capabilities for monitoring, tracing and diagnosing distributed system. It’s especially designed for microservices, cloud native and container-based architectures. 
+
+Apache SkyWalking provides a list of agents to be used for building *Java*, *.NET Core*, *PHP*, *NodeJS*, *Golang*, *LUA*, *Rust* and *C++* apps. It provides tracing, metrics analysis, alerting, service mesh observability and visualization.
+
+Apache SkyWalking is lightweight and scalable. It can be easily set up as a *self-managed* APM tool within an on-premises data center. This avoids leasing customer data to third party services and matches well with the strict security restriction in HPE GreenLake for Private Cloud Enterprise environment. 
+
+## Set up Apache SkyWalking for Application Monitoring
+
+I will take the approach to setting up the Apache SkyWalking as a *self-managed* APM tool within the Kubernetes cluster created in HPE GreenLake for Private Cloud Enterprise. This mainly takes into account the security concerns in HPE GreenLake product environment. 
```

Comment:
You are absolutely right. HPE GreenLake provides excellent security. By setting up the APM tool within this environment, rather than outside, it can benefit from such security.

Reply:
Okay, so it should be rewritten to express what you just said, that by setting up the APM tool within this environment, it can benefit from the security features of HPE GreenLake.

Reply:
Thanks Dale.  It's been addressed in the new patch.